packages/application-shell/src/components/custom-view-context/custom-view-context.tsx

```diff
@@ -0,0 +1,62 @@
+import { createContext, type ReactNode, useContext, useMemo } from 'react';
+import {
+  type TCustomView,
+  TCustomViewStatus,
+  TCustomViewType,
+} from '../../types/generated/settings';
+
+type TCustomViewContext = {
+  config: TCustomView;
+};
+
+type TCustomViewContextProviderProps = {
+  config: TCustomView;
+  children: ReactNode;
+};
+
+const CustomViewContext = createContext<TCustomViewContext>({
```

Comment:
Is this something supposed to be consumed by a Custom View?

Asking because the Custom View will likely have this data anyway exposed via a global variable like `window.app`, which is loaded from the config file.

Do you see other use cases that would benefit from using this context?

Reply:
The way I see it, `window.app` will not contain all the Custom View configuration and I'm also using this context in a follow-up PR to store another piece of information.
Perhaps we can discuss about it later in that PR.